File: Tutorials and Problems/P2a/Solutions a/LunchList1a/LunchList1a/app/src/main/java/org/feup/apm/lunchlist1a/MainActivity.java
package org.feup.apm.lunchlist1a;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    ActionBar bar = getSupportActionBar();
    if (bar != null) {
      bar.setIcon(R.drawable.rest_icon);
      bar.setDisplayShowHomeEnabled(true);
    }
    setContentView(R.layout.activity_main);
    findViewById(R.id.bt_save).setOnClickListener((View v)->onBtSaveClick());
  }

  void onBtSaveClick() {
    Restaurant r = new Restaurant();
    r.setName(((EditText)findViewById(R.id.ed_name)).getText().toString());
    r.setAddress(((EditText)findViewById(R.id.ed_address)).getText().toString());
  }

}

File: Tutorials and Problems/3.1. The Lunch Restaurants App/LunchList1/LunchList1a/app/src/main/java/org/feup/apm/lunchlist1a/MainActivity.kt
package org.feup.apm.lunchlist1a

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    supportActionBar?.setIcon(R.drawable.rest_icon)
    supportActionBar?.setDisplayShowHomeEnabled(true)
    setContentView(R.layout.activity_main)
    findViewById<Button>(R.id.bt_save).setOnClickListener { _ -> onBtSaveClick() }
  }

  private fun onBtSaveClick() {
    val r = Restaurant()
    r.name = findViewById<EditText>(R.id.ed_name).text.toString()
    r.address = findViewById<EditText>(R.id.ed_address).text.toString()
  }
}

class Restaurant {
  var name = ""
  var address = ""
}